### PR TITLE
Improve run time of data construction

### PR DESCRIPTION
**Pull request by pverscha**
This PR improves the execution time of the `run.sh` script from an average of 1213s to 792s for constructing the SwissProt database by making a few changes:

* First of all, @bmesuere and I ported the `FunctionalAnalysisPeptides.java` script to JavaScript (NodeJS). By doing so, we achieved a very respectable speedup with our real world test data (from 168s to 67s on the same machine).
* Next, I parallelised some of the steps in the build-process to better leverage the capabilities of multi-core CPU's. Computing the functional annotations for the original and equalized peptides, for example, can be performed in parallel. This lead to another great improvement on the SwissProt test set.

I expect these changes to make a big difference on the construction time of the complete Trembl database.

The output of the new script has been compared to the output of the old build script and is verified to be identical.

### CODE REVIEW

**Review by bmesuere — APPROVED** [review]
:+1:

File: run.sh [hunk 1]
```diff
@@ -59,15 +59,13 @@ java_() {
 gz() {
 	fifo="$TMP/$(uuidgen)-$(basename "$1")"
 	mkfifo "$fifo"
-	echo "$fifo"
```
Comment (ninewise) [inline run.sh:62]: I thought this line was crucial to get the name of the fifo as output of this function, so the `> "$(gz ...)"` construct works. Same for the `guz`.

File: run.sh [hunk 2]
```diff
@@ -472,19 +470,31 @@ database)
...
-	join_original_pepts_and_entries
+	join_equalized_pepts_and_entries &
+	pid1=$!
+	join_original_pepts_and_entries &
+	pid2=$!
+	wait $pid1
+	wait $pid2
 	number_sequences
-	calculate_equalized_lcas
+    calculate_equalized_lcas &
```
Comment (ninewise) [inline run.sh:480]: Inconsistent indent type, the rest of the file uses a single tab.

File: run.sh [hunk 3]
```diff
@@ -472,19 +470,31 @@ database)
...
+	pid1=$!
+	join_original_pepts_and_entries &
+	pid2=$!
+	wait $pid1
+	wait $pid2
 	number_sequences
-	calculate_equalized_lcas
+    calculate_equalized_lcas & 
+	pid1=$!
+    calculate_original_lcas &
```
Comment (ninewise) [inline run.sh:482]: Tab

File: .devcontainer/Dockerfile [hunk 4]
```diff
@@ -3,7 +3,7 @@ FROM mysql:5.7
 LABEL maintainer="Pieter Verschaffelt <pieter.verschaffelt@ugent.be>"
 
 RUN mkdir -p /usr/share/man/man1
-RUN apt-get update && apt-get install -y default-jdk git dos2unix maven make wget unzip expect gawk binutils gcc libssl-dev
+RUN apt-get update && apt-get install -y default-jdk git dos2unix maven make wget unzip expect gawk binutils gcc libssl-dev uuid-runtime pv
```
Comment (ninewise) [inline .devcontainer/Dockerfile:6]: Add node as an additional dependency? (Also remove make, I forgot to remove it in the PR where I removed the dependency on make.)

**Review by ninewise — CHANGES_REQUESTED** [review]
Could you add `module load node` to the `make-on-hpc.sh` script as well, since this PR makes it an additional dependency.

File: .devcontainer/Dockerfile (hunk 4 above)
Comment (pverscha) [reply .devcontainer/Dockerfile:6]: * make is still required for building curl ourselves (this is required due to a bug in the curl-version that's bundled with the mysql-docker)
* I will add Node as a dependency :)

**Review by ninewise — APPROVED** [review]

**Pull request merged**